airflow/providers/cncf/kubernetes/operators/custom_object_launcher.py

```diff
@@ -0,0 +1,333 @@
+        if self.config_map_mounts:
+            vols, vols_mounts = convert_configmap_to_volume(self.config_map_mounts)
+            self.volumes.extend(vols)
+            self.volume_mounts.extend(vols_mounts)
+        if self.from_env_config_map:
+            self.env_from.extend([convert_configmap(c_name) for c_name in self.from_env_config_map])
+        if self.from_env_secret:
+            self.env_from.extend([convert_secret(c) for c in self.from_env_secret])
+
+
+class SparkResources:
+    """spark resources"""
+
+    def __init__(
+        self,
+        driver=None,
+        executor=None
+    ):
+        self.default = {
+            'gpu': {'name': None, 'quantity': 0},
+            'cpu': {'request': None, 'limit': None},
+            'memory': {'request': None, 'limit': None}
+        }
+        self.driver = deepcopy(self.default)
+        self.executor = deepcopy(self.default)
+        if driver:
+            self.driver.update(driver)
+        if executor:
+            self.executor.update(executor)
+        self.convert_resources()
+
+    @property
+    def resources(self):
+        """Return job resources"""
+        return {'driver': self.driver_resources, 'executor': self.executor_resources}
+
+    @property
+    def driver_resources(self):
+        """Return resources to use"""
+        driver = {}
+        if self.driver['cpu'].get('request'):
+            driver['cores'] = self.driver['cpu']['request']
+        if self.driver['cpu'].get('limit'):
+            driver['coreLimit'] = self.driver['cpu']['limit']
+        if self.driver['memory'].get('limit'):
+            driver['memory'] = self.driver['memory']['limit']
+        if self.driver['gpu'].get('name') and self.driver['gpu'].get('quantity'):
+            driver['gpu'] = {'name': self.driver['gpu']['name'], 'quantity': self.driver['gpu']['quantity']}
+        return driver
+
+    @property
+    def executor_resources(self):
+        """Return resources to use"""
+        executor = {}
+        if self.executor['cpu'].get('request'):
+            executor['cores'] = self.executor['cpu']['request']
+        if self.executor['cpu'].get('limit'):
+            executor['coreLimit'] = self.executor['cpu']['limit']
+        if self.executor['memory'].get('limit'):
+            executor['memory'] = self.executor['memory']['limit']
+        if self.executor['gpu'].get('name') and self.executor['gpu'].get('quantity'):
+            executor['gpu'] = {'name': self.executor['gpu']['name'], 'quantity': self.executor['gpu']['quantity']}
+        return executor
+
+    def convert_resources(self):
+        if isinstance(self.driver['memory'].get('limit'), str):
+            if 'G' in self.driver['memory']['limit'] or 'Gi' in self.driver['memory']['limit']:
+                self.driver['memory']['limit'] = float(self.driver['memory']['limit'].rstrip('Gi G')) * 1024
+            elif 'm' in self.driver['memory']['limit']:
```

Comment:
Any possibility that user could use 'M' instead of 'm'? 'g' instead of 'G'. Should we provide some better experience in these case?